manifests/server/pg_hba_rule.pp

```diff
@@ -1,19 +1,19 @@
 # This resource manages an individual rule that applies to the file defined in
 # $target. See README.md for more details.
 define postgresql::server::pg_hba_rule(
-  $type,
-  $database,
-  $user,
-  $auth_method,
-  $address     = undef,
-  $description = 'none',
-  $auth_option = undef,
-  $order       = '150',
+  Enum['local', 'host', 'hostssl', 'hostnossl'] $type,
+  String $database,
+  String $user,
+  String $auth_method,
```

Comment:
In the validation code there's some regex validation with ```validate_re```. I believe you can simplify that to ```assert_type(Enum[$allowed_auth_methods], $auth_method)``` but I'm not 100% sure.